alternator/conditions.cc

```diff
@@ -169,17 +169,22 @@ static bool verify_expected_one(const rjson::value& condition, const rjson::valu
         if (!attribute_value_list || !attribute_value_list->IsArray()) {
             throw api_error("ValidationException", "With ComparisonOperator, AttributeValueList must be given and an array");
         }
+        using check_t = bool(*)(const rjson::value*, const rjson::value&);
+        static const std::unordered_map<comparison_operator_type, check_t> checks{
+            {comparison_operator_type::EQ, &check_EQ},
+            {comparison_operator_type::BEGINS_WITH, &check_BEGINS_WITH},
```

Comment:
I'm sorry, I don't understand the **goal** of this patch.
How is it better than a switch()? It's not shorter than the switch - you still have one line for every one of the operators. It's not faster than a switch. It's not clearer. So why change this?
Especially since you even **left** the switch, for IN and BETWEEN. So why not just leave it for all of them? What am I missing?

Reply:
This pattern has been agreed upon in [prior review](https://github.com/scylladb/scylla/pull/5110#discussion_r329104078).  It is used [elsewhere](https://github.com/scylladb/scylla/blob/master/alternator/conditions.cc#L36) in this code, too.  It is 3x shorter (per operator) than the code it replaces, because it allows many operators to reuse the same `switch` case.